ui/src/timeMachine/components/TimeMachineFluxEditor.tsx

```diff
@@ -56,31 +56,58 @@ const TimeMachineFluxEditor: FC<Props> = ({
     onSetActiveQueryText(editorInstance.getValue())
   }
 
+  const getInsertLineNumber = (
+    currentLineNumber: number,
+    scriptLines: string[]
+  ): number => {
+    const currentLine =
+      scriptLines[currentLineNumber] || scriptLines[scriptLines.length - 1]
+
+    // Insert on the current line if its an empty line
+    if (!currentLine.trim()) {
+      return currentLineNumber
+    }
+
+    return currentLineNumber + 1
+  }
+
   const handleInsertFluxFunction = (func: FluxToolbarFunction): void => {
     const p = editorInstance.getPosition()
+    const scriptLines = activeQueryText.split('\n')
+
+    const insertLineNumber = getInsertLineNumber(p.lineNumber, scriptLines)
+
     // sets the range based on the current position
     let range = new window.monaco.Range(
-      p.lineNumber,
-      p.column,
-      p.lineNumber,
-      p.column
+      insertLineNumber, // the row beneath the cursor
```

Comment:
is this comment true? if there's nothing on the line the cursor is on, it will remain on the current line. i think you should remove the comment

Reply:
It is true.

Reply:
i think the term "the row beneath" is confusing in this context. it could mean "the line under the line where the cursor is" e.g. cursor is on line 1, place it on line 2 (how i originally interpreted it) or "directly where the cursor is, directly under it" e.g. cursor is on line 1, place the function on line 1 directly under the cursor. this comment added to my confusion, rather than clarifying my understanding.

Reply:
Gotcha. I'll update the comment to reflect the former string that you said

Reply:
Since that particular property is only determining the row (and not the column), it's only concerned with the `row` context. In that sense, the column has no bearing on this value so it's only referring to the row